lib/hotel_block.rb

```diff
@@ -0,0 +1,96 @@
+require 'date'
+# # require_relative 'hotel.rb'
+
+module Booking
+  class Block
+
+    attr_reader :checkin_date, :checkout_date, :requested_rooms, :discounted_rate, :party_name, :room_block, :block_of_rooms, :all_rooms, :max_rooms, :room_numbers, :hotel, :all_reservations
+    
+    def initialize(checkin_date, checkout_date, requested_rooms, discounted_rate, hotel: nil)
+    @checkin_date = checkin_date
+    @checkout_date = checkout_date 
+    @block_of_rooms ||= []
+    @hotel ||= hotel
+    @available_rooms = @hotel.check_availability(checkin_date, checkout_date)
```

Comment:
This syntax says that every instance of `HotelBlock` should keep a record of all available rooms in an instance variable `@available_rooms`... and that this value should come from everything `@hotel` says is available. I would say that this seems weird that a `Block` needs to know all available rooms...